flake.nix

```diff
@@ -8,14 +8,18 @@
     haskell-nix.follows = "haskell-backend/haskell-nix";
```

Comment:
I noticed that we are testing with a 5.6 version of K here... but upgrading seems to cause issues with the index-state being too new so I left that for a follow-up.